### ISSUE TITLE
Gray Screen Display and Color Inaccuracy when Enabling HDR High Dynamic Range Option for Video Playback.

### ISSUE THREAD

**Issue by dfvips** [iina/iina#4440]
**System and IINA version:**

- macOS 13.4
- IINA 1.3.1

**Expected behavior:**
Properly decoding and playing the colors of HDR content when playing HDR videos with the 'HDR High Dynamic Range' option enabled.
The image below shows the result after disabling the HDR High Dynamic Range option, which appears relatively normal. However, upon comparing it with the colors rendered by QuickTime decoding, the colors in IINA also seem to be somewhat abnormal.

<img width="1438" alt="image" src="https://github.com/iina/iina/assets/21048630/312c0aa8-6fe5-4c64-a49d-ee76bfc0140a">
<img width="1440" alt="image" src="https://github.com/iina/iina/assets/21048630/dc23505f-384d-4151-ad8f-095876645839">

**Actual behavior:**
<img width="1213" alt="image" src="https://github.com/iina/iina/assets/21048630/60de6163-4e0f-45bf-a0df-0c1682e08f90">
<img width="1423" alt="image" src="https://github.com/iina/iina/assets/21048630/ce4806f9-2ecb-4d1f-b8c4-ef682406faa0">

When playing HDR videos with the 'HDR High Dynamic Range' option enabled, the player turns gray and it is not possible to watch the video with real colors properly，but when taking screenshots or recording the screen, it appears in color.

**How often does this happen?**
Always.

**Comment by dfvips** [comment]
https://github.com/iina/iina/assets/21048630/e36d6172-5c66-4710-8622-252666f50a7f
This is an HDR video intended for testing purposes.

**Comment by low-batt** [comment]
Playing the attached sample on my MacBookPro18,2 with the M1 chip did not reproduce the grey screen for me. The picture did look too bright compared to QuickTime.

Are you using a Mac with the M2 chip? There are known serious problems playing HDR videos with IINA under the M2 chip, although I have not heard of them causing a grey screen.

The latest development version of IINA contains a fix to HDR support. Users have reported this corrects the problems experienced when running on the M2. The new version also adds support for tone mapping which can be enabled in IINA's settings under the `Video/Audio` tab. Playing the sample video with the latest IINA nightly build downloaded from [here](https://nightly.iina.io/) the picture looked pretty close to what is seen when played with QuickTime.

As can be seen from the pinned issue IINA is very close to releasing a new version with updated HDR support.

**Comment by dfvips** [comment]
> Playing the attached sample on my MacBookPro18,2 with the M1 chip did not reproduce the grey screen for me. The picture did look too bright compared to QuickTime.
> 
> 
> 
> Are you using a Mac with the M2 chip? There are known serious problems playing HDR videos with IINA under the M2 chip, although I have not heard of them causing a grey screen.
> 
> 
> 
> The latest development version of IINA contains a fix to HDR support. Users have reported this corrects the problems experienced when running on the M2. The new version also adds support for tone mapping which can be enabled in IINA's settings under the `Video/Audio` tab. Playing the sample video with the latest IINA nightly build downloaded from [here](https://nightly.iina.io/) the picture looked pretty close to what is seen when played with QuickTime.
> 
> 
> 
> As can be seen from the pinned issue IINA is very close to releasing a new version with updated HDR support.

Yes, I am using a Mac with the M2 chip. I appreciate your explanation about the known issues with playing HDR videos on IINA under the M2 chip. I will download the latest development version and give it a try to see if the problem is resolved. Looking forward to the updated HDR support in the upcoming release. Thank you for your hard work!

**Comment by dfvips** [comment]
> Playing the attached sample on my MacBookPro18,2 with the M1 chip did not reproduce the grey screen for me. The picture did look too bright compared to QuickTime.
> 
> Are you using a Mac with the M2 chip? There are known serious problems playing HDR videos with IINA under the M2 chip, although I have not heard of them causing a grey screen.
> 
> The latest development version of IINA contains a fix to HDR support. Users have reported this corrects the problems experienced when running on the M2. The new version also adds support for tone mapping which can be enabled in IINA's settings under the `Video/Audio` tab. Playing the sample video with the latest IINA nightly build downloaded from [here](https://nightly.iina.io/) the picture looked pretty close to what is seen when played with QuickTime.
> 
> As can be seen from the pinned issue IINA is very close to releasing a new version with updated HDR support.

I have just reviewed the developer version and I can confirm that the grey screen issue with HDR videos has indeed been resolved. However, it seems that the video playback effect is slightly overexposed. Here is a comparison of the HDR playback effects on several commonly used professional players for Mac. You can take a look and compare. It's possible that my perception may not be entirely accurate, but it appears that other players do not exhibit overexposure in their HDR effects.

Thank you for your attention and your continued efforts in improving the HDR support in IINA.

![image](https://github.com/iina/iina/assets/21048630/857e0216-a6ec-4307-bfd9-332174bd508e)
<img width="1440" alt="image" src="https://github.com/iina/iina/assets/21048630/5832c822-dabe-4995-b57b-8559a30fa173">
<img width="1497" alt="image" src="https://github.com/iina/iina/assets/21048630/a4a27a06-5723-49e1-9b61-99a148c69b99">

**Comment by low-batt** [comment]
Glad to hear the grey screen problem is resolved.

I'm hoping @CarterLi will comment on the overexposure issue as he knows the HDR code.

**Comment by dfvips** [comment]
> Glad to hear the grey screen problem is resolved.
> 
> 
> 
> I'm hoping @CarterLi will comment on the overexposure issue as he knows the HDR code.

Thank you for your reply. In addition, I'm hoping that the HDR expert can provide further insight into the overexposure issue. I look forward to his response.

**Comment by CarterLi** [comment]
It still seems to be a tone mapping issue. Do you use Macbook Air or Pro?

**Comment by CarterLi** [comment]
Noting that it's HDR vivid, and I doubt if MPV supports it natively, or fallbacks to HDR10 silently. However it should not make much difference.

**Comment by dfvips** [comment]
> It still seems to be a tone mapping issue. Do you use Macbook Air or Pro?

I am using a MacBook Pro with the M2 chip, specifically the 13-inch model.

**Comment by dfvips** [comment]
> Noting that it's HDR vivid, and I doubt if MPV supports it natively, or fallbacks to HDR10 silently. However it should not make much difference.

Thank you for your clarification. In the case where MPV cannot provide accurate color mapping, I would like to know if IINA can implement any specific solution to address this color mapping issue.

**Comment by dfvips** [comment]
<img width="1440" alt="image" src="https://github.com/iina/iina/assets/21048630/6c2a1fa2-1ed0-43e3-8f7e-01e3dc378b35">
<img width="1440" alt="image" src="https://github.com/iina/iina/assets/21048630/aedb91e2-2a0a-4a86-a7e3-3e533b40da6a">
<img width="1440" alt="image" src="https://github.com/iina/iina/assets/21048630/19eb64cf-67c4-4515-94ab-82070ff56a63">
<img width="1440" alt="image" src="https://github.com/iina/iina/assets/21048630/5adf4c5f-b74d-4967-acf1-96a42757671a">

video for test

https://github.com/iina/iina/assets/21048630/2f02521a-7182-4aae-856f-98fbefb6a8f8

This is a side-by-side comparison image, and below is a test HDR video. Pay attention to the colors of the clothing in the picture.
